src/glum/_solvers.py

```diff
@@ -792,7 +794,12 @@ def line_search(state: IRLSState, data: IRLSData, d: np.ndarray):
         )
         # 1. Check Armijo / sufficient decrease condition.
         loss_improvement = obj_val_wd - state.obj_val
-        if mu_wd.max() < 1e43 and loss_improvement <= factor * bound:
+        if mu_wd.dtype == np.float32:
```

Comment:
Can we move this logic into the IRLSData class? Only need to compute it once (where I would expect it).